src/server/stream_family.cc

```diff
@@ -1364,6 +1502,124 @@ void StreamFamily::XAdd(CmdArgList args, ConnectionContext* cntx) {
   return (*cntx)->SendError(add_result.status());
 }
 
+void StreamFamily::XClaim(CmdArgList args, ConnectionContext* cntx) {
+  ClaimOpts opts;
+  string_view key = ArgS(args, 0);
+  opts.group = ArgS(args, 1);
+  opts.consumer = ArgS(args, 2);
+  if (!absl::SimpleAtoi(ArgS(args, 3), &opts.min_idle_time)) {
+    return (*cntx)->SendError(kSyntaxErr);
+  }
+  // Ignore negative min-idle-time
+  opts.min_idle_time = std::max(opts.min_idle_time, static_cast<int64>(0));
+  args.remove_prefix(4);
+
+  auto GetXclaimIds = [&]() {
```

Comment:
Having this lambda inline isn't helping in making this function more readable (or shorter :)
Can you move GetXclaimIds() to be a standalone function, probably in anonymous namespace?